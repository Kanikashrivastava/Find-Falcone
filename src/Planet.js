import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import FormControl from "@material-ui/core/FormControl";
import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
  select: {
    fontSize: 21,
    padding: "21px",
  },
}));

export function DestinationVehicleBook({ optionNum, planets, vehicles, handlePlanetChange, handleVehicleChange }) {
  const classes = useStyles();
  const [selectedPlanet, setSelectedPlanet] = useState("default");
  const [selectedVechile, setSelectedVechile] = useState("default");
  const isPlanetSelected = () => {
    return selectedPlanet && selectedPlanet !== "default";
  };
  const onPlanetChange = (e) => {
    const { value } = e.target;
    setSelectedPlanet(value)
    handlePlanetChange(value);
  }
  const onVehicleChange = (e) => {
    const { value } = e.target;
    setSelectedVechile(value)
    handleVehicleChange(value);
  }
  return (
    <>
      <FormControl variant="filled" className={classes.formControl}>
        <label id="demo-simple-select-label">Destination {optionNum}</label>
        <select
          id="planet-select"
          data-testId="planet-select"
          className={classes.select}
          onChange={onPlanetChange}
        >
          {[{ name: "default", distance: 0 }, ...planets].map((planet) => {
            return (
              <option
                data-testId={planet.name}
                value={planet.name}
                selected={selectedPlanet === planet.name}
              >
                {planet.name === 'default' ? 'Select Planet' : planet.name}
              </option>
            );
          })}
        </select>
        {isPlanetSelected() && (
          <RadioGroup
            aria-label="vehicles"
            name="vehicles"
            value={selectedVechile}
            onChange={onVehicleChange}
          >
            {vehicles.map((vehicle) => {
              return (
                <FormControlLabel
                  value={vehicle.name}
                  control={<Radio data-testId={vehicle.name}/>}
                  label={`${vehicle.name.toUpperCase()} (${vehicle.total_no})`}
                />
              );
            })}
          </RadioGroup>
        )}
      </FormControl>
    </>
  );
}

DestinationVehicleBook.defaultProps = {
  handlePlanetChange: () => { },
  handleVehicleChange: () => { }
}