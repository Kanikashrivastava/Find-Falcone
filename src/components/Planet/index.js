import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';


const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
  select: {
    fontSize: 21,
    padding: '21px',
  }
}));

export function DestinationVehicleBook({ optionNum, planets, vehicles }) {
  const classes = useStyles();
  const [selectedPlanet, setSelectedPlanet] = useState('default')
  const [selectedVechile, setSelectedVechile] = useState('default')
  const isPlanetSelected = () => {
    return selectedPlanet && selectedPlanet !== 'default'
  }
  return (
    <>
      <FormControl variant="filled" className={classes.formControl}>
        <label id="demo-simple-select-label">Destination {optionNum}</label>
        <select
          id="planet-select"
          data-testId="planet-select"
          value={selectedPlanet}
          className={classes.select}
          onChange={(e) => setSelectedPlanet(e.target.value)}
        >
          <option value="default">Select Planet</option>
          {planets.map((planet) => {
            return <option value={planet.name}>{planet.name}</option>
          })}
        </select>
        {isPlanetSelected() &&
          (
            <RadioGroup 
              aria-label="vehicles" 
              name="vehicles" 
              value={selectedVechile} 
              onChange={
                (e) => setSelectedVechile(e.target.value)
              }
            >
              {vehicles.map((vehicle) => {
                return (
                  <FormControlLabel value={vehicle.name} control={<Radio />} label={`${vehicle.name.toUpperCase()} (${vehicle.total_no})`} />
                )
              })}
            </RadioGroup>
          )
        }
      </FormControl>
    </>
  )
}