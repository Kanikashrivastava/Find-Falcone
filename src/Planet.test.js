import React from 'react';
import { render, cleanup, screen, prettyDOM } from '@testing-library/react';
import user from '@testing-library/user-event';
import { DestinationVehicleBook } from './Planet';
afterAll(() => {
  cleanup()
  jest.resetAllMocks()
})
test('renders planets options', async () => {
    const props = {
        optionNum: 1,
        planets: [{ name: 'Planet 1' }, { name: 'Planet 2' }],
        vehicles: [
            { name: 'Vehicles 1', total_no: 1 },
            { name: 'Vehicles 2', total_no: 2 }
        ]
    }
    const utils = render(<DestinationVehicleBook {...props} />);
    const { asFragment } = utils
    user.click(screen.getByTestId("planet-select"))
    user.selectOptions(screen.getByTestId(/planet-select/), [props.planets[0].name])
    expect(asFragment()).toMatchSnapshot()
});

test(`should trigger handle planet change when user selects an option`, async () => {
    const props = {
        optionNum: 1,
        planets: [{ name: 'Planet 1' }, { name: 'Planet 2' }],
        vehicles: [
            { name: 'Vehicles 1', total_no: 1 },
            { name: 'Vehicles 2', total_no: 2 }
        ],
        handlePlanetChange: jest.fn()
    }
    const utils = render(<DestinationVehicleBook {...props} />);
    user.click(utils.getByTestId("planet-select"))
    user.selectOptions(utils.getByTestId(/planet-select/), [props.planets[0].name])
    expect(props.handlePlanetChange).toHaveBeenCalledTimes(1)

})

test(`should trigger handle vehicle change when user selects a vehicle`, async () => {
    const props = {
        optionNum: 1,
        planets: [{ name: 'Planet 1' }, { name: 'Planet 2' }],
        vehicles: [
            { name: 'Vehicles 1', total_no: 1 },
            { name: 'Vehicles 2', total_no: 2 }
        ],
        handleVehicleChange: jest.fn()
    }
    const utils = render(<DestinationVehicleBook {...props} />);
    user.click(utils.getByTestId("planet-select"))
    user.selectOptions(utils.getByTestId(/planet-select/), [props.planets[0].name])
    user.selectOptions(utils.getByTestId(/Vehicles 1/), [props.vehicles[0].name])
    // console.log(prettyDOM(utils.getByLabelText(/vehicles 2/)),)
    expect(props.handleVehicleChange).toHaveBeenCalledTimes(1)

})
