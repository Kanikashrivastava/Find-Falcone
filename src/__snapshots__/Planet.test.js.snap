// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders planets options 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root makeStyles-formControl-1"
  >
    <label
      id="demo-simple-select-label"
    >
      Destination 1
    </label>
    <select
      class="makeStyles-select-3"
      data-testid="planet-select"
      id="planet-select"
    >
      <option
        data-testid="default"
        value="default"
      >
        Select Planet
      </option>
      <option
        data-testid="Planet 1"
        value="Planet 1"
      >
        Planet 1
      </option>
      <option
        data-testid="Planet 2"
        value="Planet 2"
      >
        Planet 2
      </option>
    </select>
    <div
      aria-label="vehicles"
      class="MuiFormGroup-root"
      role="radiogroup"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
          data-testid="Vehicles 1"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-24"
              name="vehicles"
              type="radio"
              value="Vehicles 1"
            />
            <div
              class="PrivateRadioButtonIcon-root-37"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-38"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          VEHICLES 1 (1)
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
          data-testid="Vehicles 2"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-24"
              name="vehicles"
              type="radio"
              value="Vehicles 2"
            />
            <div
              class="PrivateRadioButtonIcon-root-37"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-38"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          VEHICLES 2 (2)
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
